### ISSUE TITLE
[test] Add test for new assert in set_goal method

### ISSUE THREAD

**Issue by EmmanuelAR** [web3wagers/gostarkme#258]
### This issue will be part of ODHack10, please apply via Onlydust app
- Add new test for the new assert admins in set_goal in contracts/src/fund.cairo.
- This issue can be work once the #254  is done
- Please read [contributors guide](https://github.com/web3wagers/gostarkme/blob/dev/CONTRIBUTORS_GUIDE.md) before asking for an issue.

**Comment by BrunoAmbricca** [comment]
I would like to work on this issue

**Comment by Emmanex01** [comment]
Can I take this issue?

**Comment by Gerson2102** [comment]
---

### Hey! I'm Gerson. 👋

Member of **Dojo Coding**. I have contributed before to other projects like:

- `kakarot-rpc`
- `ef-tests`
- `Raito`
- `Starknet-Phone`

Can I work on this issue?

Check my OnlyDust profile, I've contributed to many projects: [Profile](https://app.onlydust.com/u/Gerson2102)

---

### Plan to Solve the Issue:

1. **Study the codebase** to understand the current structure and functionality.
2. **Understand the problem** that needs to be solved by thoroughly reading the issue details.
3. **Leverage all provided resources** within the issue to get a better grasp of the task at hand.
4. If I encounter roadblocks, I will **ask questions ASAP** in the Telegram group to move forward effectively.
5. **I will open a PR** and wait for ur feedback

---

**Comment by No-bodyq** [comment]
I'd like to handle this task.

**Comment by Unickhub** [comment]
Is it okay if I take this?

**Comment by mimisavage** [comment]
I'd like to handle this task.

**Comment by ncoquelet** [comment]
Is this issue still available?

# Presentation

Hello, I'm senior developer, comfortable with Typescript, React/Vite/Next, testing, CI/CD, docs and I'm a beginner on Cairo (starklings 100% and NodeGuardians “Entering Cairo”, “Thinking in Cairo”, “Vanity Address”, and I continue).
I'm rigorous and diligent, I'm used to contributing my skills to projects and can move really quickly.

# How do I resolve this issue?

I will Implement specifications.

- add new test as mentioned
- can take #254 too to go faster

Contributors guide checked !

**Comment by GideonBature** [comment]
Can I be assigned to this?

My name is Gideon Bature, a Software Engineer with experience building Starknet based smart contracts, using Cairo. If given this issue, I should be able to send my first PR within 48 hrs right after issue #254 is completed. 

Thank you!

**Comment by Benjtalkshow** [comment]
Can I take this issue?

**Comment by ryzen-xp** [comment]
Hi, I am proficient Blockchain developer  expert in Cairo , Rust and Solidity  and also in web development.
I have  contributed many stark net project and more  .   please /assign  this issue .
Thank You

**Comment by ShantelPeters** [comment]
I'd like to take this issue.

**Comment by Michaelkingsdev** [comment]
May I take care of this?

**Comment by SoarinSkySagar** [comment]
May I work on this issue once the mentioned issue is resolved, @EmmanuelAR?

I am an experienced cairo dev and a regular contributor at @keep-starknet-strange.

eta: 3 days

**Comment by Kom02** [comment]
Hi, My name is Kevin Obando, a full stack junior developer with 2 years of experience from Costa Rica. I am starting in this community and I want to help in this project.

**Comment by NueloSE** [comment]
I’m interested in this one.

Am experienced in writing and testing smart contract using cairo.  ETA <= 24hrs

**Comment by ikezuby2012** [comment]
hello, can i be assigned this issue?

my name is nnubia nzube, i'm a software developer with experience in writing smart contract using cairo, i would like to write the test for the contract, my ETA is within 48 hours.

**Comment by saimeunt** [comment]
Hey, I'm a returning contributor having previously worked on testing gostarkme, I think this can be completed in half a day.

**Comment by Gianfranco99** [comment]
Hello, I can solve this issue easily, I have already been contributing to this project and I would like to continue being part of it!

**Comment by petersdt** [comment]
Mind if I take this issue?

Experienced in testing and writing proveable smart contract using cairo and solidity.

For the above task based on the implementation from issue#254 i would implement a uint test to cover all necessary edge case with the implementation of set_name method.

this would be nice to have

**Comment by tnv1** [comment]
I'd like to handle this task. Thank you!

**Comment by mimiprosper** [comment]
Let me try this one!  Am a web 2 frontend developer, a solidity smart contract developer/auditor and a cairo smart contract developer. Am a graduate of Starknet Cairo Bootcamp. I have successfully done tasks assigned to me in OnlyDust. If assigned this i would complete it within 4 days.

**Comment by bruhhgnik** [comment]
i can do this, eta: 24 hours

**Comment by wheval** [comment]
I’d like to resolve this.

**Comment by hossamanany** [comment]
Can I take this from here?

**Comment by Thibrac** [comment]
I'd like to take this issue.

**Comment by Aditya-040** [comment]
May I be assigned to this?

**Comment by renzobanegass** [comment]
Hi! I have been assigned issues on this project before but couldn't complete it, would love to give it a try again. I'm a Software Developer with over 5+ years of experience and a member of Dojo Coding, I think I can provide a fast PR as soon as the #254 issue is done

**Comment by Rayguti** [comment]
Description:
This issue involves creating a new test for the assert_admins function in the set_goal method within contracts/src/fund.cairo. The task is dependent on issue #254 and aims to ensure comprehensive testing for the updated functionality. Contributors are advised to read the contributor's guide before proceeding.

Why I’m Interested:
With experience in smart contract development and testing, I am eager to contribute to ODHack10.0 by adding meaningful tests to the project. This issue provides an excellent opportunity to refine my skills in Cairo and collaborate with the Onlydust community. My background in writing structured and reliable test cases ensures I can effectively deliver this feature while maintaining high-quality standards.

Proposed Approach:

    Review Dependencies:
        Examine changes introduced in issue #254 to understand the updates in the set_goal method.
        Analyze the behavior and expected outcomes of the assert_admins function.

    Design Test Cases:
        Develop test scenarios to verify:
            Proper functionality when valid admins are involved.
            Handling of unauthorized users or invalid admin roles.
            Edge cases, such as boundary conditions for admin privileges.

    Implementation:
        Use Cairo’s testing framework to write and run the new test cases.
        Ensure the tests align with the structure and conventions of the project’s existing test suite.

    Validation:
        Run tests in different scenarios to validate the robustness of the assert_admins functionality.
        Ensure meaningful error messages and debug information are included for failed cases.

    Collaboration:
        Coordinate with contributors of issue #254 to ensure a seamless handoff.
        Follow up with maintainers for feedback on test implementation.

Outcome:
A robust and comprehensive test for the assert_admins function will be added to the set_goal method, ensuring reliability and proper functionality of the feature.

**Comment by abdegenius** [comment]
I'd love to give this a go.

**Comment by adrianvrj** [comment]
hi @renzobanegass any updates on this? :)

**Comment by renzobanegass** [comment]
Hi! @adrianvrj I'll be creating a PR today
